About copy

Bob-IT opened the pull request:
This was what I was think for coping the About configuration text in one go. I have added some text and copy icon that can be clicked on to copy the setting to the clipboard. The first image below shows the about dialog and the second how it was pasted to a text file.

<img width="484" height="537" alt="Screenshot_2025-11-24_14-46-02" src="https://github.com/user-attachments/assets/7175c2a4-c85f-4c66-a311-dc714e846ba9" />
<img width="397" height="189" alt="about-copy" src="https://github.com/user-attachments/assets/6440b94b-0ff4-410f-84cd-2f883d768f7e" />

The second commit adds the operating system to the dialog as seen below...
<img width="484" height="554" alt="Screenshot_2025-11-24_15-00-00" src="https://github.com/user-attachments/assets/aab46e6e-aff1-417a-8773-43ecd117b909" />

Bob-IT committed:
Add ability to copy About dialog configuration
settings to the clipboard so they can all be
pasted in one go.

jralls commented:
I guess including the OS is harmless, but it seems unnecessary to me. Surely the user knows what OS they're running.

jralls commented on gnucash/gnome-utils/gnc-main-window.cpp:
```diff
@@ -5276,6 +5276,65 @@ link_button_cb (GtkLinkButton *button, gpointer user_data)
+                       gpointer user_data)
+{
+    GtkClipboard *clipboard = gtk_clipboard_get_default (gdk_display_get_default());
+    GList *entries = nullptr;
```
This is a C++ file, use a `std::string` to collect the entries and pass `entries.c_str()` to `gtk_clipboard_set_text`.

jralls reviewed (changes requested):
I think it would be better to replace the labels and buttons with a [GtkTextView](https://docs.gtk.org/gtk3/class.TextView.html) using markup to make the clickable links. That would remove the need for duplication and let the user select what they want to copy instead of having to copy the whole thing and delete the bits they don't want after pasting.

christopherlam commented:
> I guess including the OS is harmless, but it seems unnecessary to me. Surely the user knows what OS they're running.

I haven't a major opinion on having the OS... perhaps my main motivation for exposing the env paths was to assist in unusual packages. The user may run on flatpak or docker on ubuntu, and if the envs are deep within the packages, exposing the paths would be useful.

In my real-life job, I've had the pleasure of using weird remote desktop apps whereby I'd log into a win10 desktop somewhere, use it proficiently, launch a different app, use it natively, and find that printing was broken. The cause was, of course, the different app was presented locally with minimal delays, but was running from another windows server somewhere, and didn't access the local (i.e. "remote") printers.

How would the OS be shown if run within WSL2 on Win11?

jralls commented:
> How would the OS be shown if run within WSL2 on Win11?

When run in an Arch Linux WSL2 VM it says Arch Linux. You'd expect that, the OS inside a VM doesn't know it's not on a dedicated computer.

I wasn't able to get a flatpak built with this PR incorporated but based on the code I'd expect it to return the same as a normal build. We'd still be able to tell it's a flatpak from the GNC_BIN path being /app/bin.

bhardwajs commented:
Since the OS name is coming from build time and inside WSL `G_OS_WIN32` is not defined, it would be the Linux disto. Would you like me to try building the branch on my WSL instance?

Bob-IT commented:
> I guess including the OS is harmless, but it seems unnecessary to me. Surely the user knows what OS they're running.

Yes they should, it was just for convenience, you see a lot of requests asking for version and machine OS. This was just a thought about coping the top section and pasting in email, bug or what ever.

Bob-IT commented:
With a GtkTextview this is what I have come up with, I have left the grid in for comparison. Using link buttons in the grid you had tooltips for the links, is that required for the GtkTextview...

<img width="484" height="656" alt="Screenshot_2025-11-26_14-31-32" src="https://github.com/user-attachments/assets/5e2d0eee-f073-4cd5-97bd-bcb1bf5a4097" />

jralls commented:
I can't think of any good use for the tooltips. Do you think that it would look better if you made the background blend in with the rest of the dialog?

Bob-IT committed:
Add CSS to make the About dialog text view
background the same as the rest of the dialog.
This can be over ridden by the users own CSS file.

Bob-IT commented:
Changes required for the About dialog to appear as above with a textview widget. The last commit changes the background to match rest of the dialog which can be over ridden by the user. This was the only way I could get a good result, tried using the same method as with the tab colors, getting the background color of the dialog and creating some CSS to apply which did work but it then upset the selection colors.

jralls commented on gnucash/gnome-utils/gnc-gnome-utils.c:
```diff
@@ -171,6 +171,53 @@ gnc_add_css_file (void)
+{
+    NSString *operatingSystemVersionString = [[NSProcessInfo processInfo] operatingSystemVersionString];
+
+    return g_strdup_printf ("Mac %s",[operatingSystemVersionString UTF8String]);
```
s/Mac/macOS/

Bob-IT replied:
dropped

jralls commented on gnucash/gnome-utils/gnc-gnome-utils.c:
```diff
@@ -171,6 +171,53 @@ gnc_add_css_file (void)
+    if (dwBuild >= 22000 && dwMajorVersion == 10)
+        dwMajorVersion = 11;
+
+    return  g_strdup_printf ("Windows Version %ld.%ld (%ld)",
```
s/Windows/Microsoft Windows/

Bob-IT replied:
dropped

jralls commented:
Aside from the OS name changes I like the look though I'm still not convinced that the OS info adds anything useful.

Bob-IT commented:
I forgot to account for using Adwaita-dark theme. Tested on Linux and Windows and looks OK.
Will push this tomorrow.
